tcp/tcp.go

```diff
@@ -83,6 +92,17 @@ func Listen(ctx context.Context, address string, handle func(net.Conn), handleEr
 	}
 }
 
+// ListenerWithAssignedPort creates a new listener on a random port assigned by
```

Comment:
I'm not sure if the `ip` parameter should be of type `net.IP`. I think `ListenWithAssignedPort` should accept it as a string (for uniformity) since `Listen` takes in the address as a string as well.

Reply:
Fixed in b6e78b1.